test/workers/delete_account_hierarchy_worker_test.rb

```diff
@@ -43,19 +43,19 @@ def setup
     DeletePaymentSettingHierarchyWorker.expects(:perform_later).with(provider.payment_gateway_setting, anything, 'destroy')
     DeleteObjectHierarchyWorker.expects(:perform_later).with(api_docs_service, anything, 'destroy')
 
-    perform_enqueued_jobs { DeleteAccountHierarchyWorker.perform_now(provider) }
+    perform_enqueued_jobs(only: DeleteAccountHierarchyWorker) { DeleteAccountHierarchyWorker.perform_now(provider) }
   end
 
   test 'does not perform if wrong state' do
     provider.update_column(:state, 'approved')
 
-    perform_enqueued_jobs { DeleteAccountHierarchyWorker.perform_now(provider) }
+    perform_enqueued_jobs(only: DeleteAccountHierarchyWorker) { DeleteAccountHierarchyWorker.perform_now(provider) }
```

Comment:
This is cheating 😝 It should be:
```
provider.update_column(:state, 'approved')

perform_enqueued_jobs(only: [DeleteAccountHierarchyWorker, DeletePlainObjectWorker]) { DeleteAccountHierarchyWorker.perform_now(provider) }

assert provider.reload
```

Reply:
Or even better if it is `perform_enqueued_jobs(except: SphinxIndexationWorker) { DeleteAccountHierarchyWorker.perform_now(provider) }` 🤔 It really has to be the same as the one in the test below 😄